home/management/commands/updatecourses.py

```diff
@@ -0,0 +1,162 @@
+            print(f"Working on courses for {semester.name()}...")
+
+            while course_data:
+                # for every course taught during `semester`...
+                for umdio_course in course_data:
+                    course = self.courses.filter(name=umdio_course['course_id'].strip("\n\t\r ")).first()
+
+                    # if we don't have the course, create it.
+                    if not course:
+                        course = Course(
+                            name=umdio_course['course_id'].strip("\n\t\r "),
+                            department=umdio_course['dept_id'].strip("\n\t\r "),
+                            course_number=umdio_course['course_id'].strip("\n\t\r ")[4:],
+                            title=umdio_course['name'].strip("\n\t\r "),
+                            credits=umdio_course['credits'].strip("\n\t\r "),
+                            description=umdio_course["description"].strip("\n\t\r ")
+                        )
+
+                        course.save()
+                        self.total_num_new_courses += 1
+
+                    print(course)
+                    # collect all the professors that taught this course during `semester`
+                    self._professors(course, semester)
+
+                kwargs["page"] += 1
+                course_data = requests.get("https://api.umd.io/v1/courses", params=kwargs).json()
+
+        print(f"\n** New Courses Created: {self.total_num_new_courses} **")
+        print(f"** New Professors Created: {self.total_num_new_professors} **")
+
+        runtime = datetime.now() - t_start
+        print(f"Runtime: {round(runtime.seconds / 60, 2)} minutes")
+
+    def _professors(self, course: Course, semester: Semester):
+        kwargs = {"course_id": course.name}
+        umdio_professors = requests.get("https://api.umd.io/v1/professors", params=kwargs).json()
+
+        # if no professors were found for `course`, exit function.
+        if isinstance(umdio_professors, dict) and 'error_code' in umdio_professors.keys():
+            return
+
+        # for every professor that taught `course`...
+        for umdio_professor in umdio_professors:
+            professor_name = umdio_professor['name'].strip("\n\t\r ")
+            if re.search("instructor:?\s*tba", professor_name.lower()):
+                continue
+
+            professor = self.non_rejected_professors.filter(name=professor_name)
+
+            # if there's only one matching professor, use that professor.
+            if professor.count() == 1:
+                professor = professor.first()
+            else:
+                alias = self.aliases.filter(alias=professor_name)
+
+                # if there's more than one matching professor but
+                # we have an alias that narrows the query down to one,
+                # use the professor associated with that alias.
+                if professor.count() > 1 and alias.count() == 1:
+                    professor = alias.first()
```

Comment:
I would hope that `alias.count()` is always one. If we have two aliases for the same name, that's a problem. We should add a `unique=True` requirement to the alias' name field.